### ISSUE TITLE
Hovering embedded table header cell crashes the app

### ISSUE THREAD

**Issue by rfricz** [handsontable/handsontable#10359]
### Description

When the data contains embedded tables and `manualColumnResize: true`, hovering a visible TH cell in the data throws an error with the following stack trace:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'start')
    at ManualColumnResize.setupHandlePosition (manualColumnResize.mjs:348:49)
    at ManualColumnResize.onMouseOver (manualColumnResize.mjs:414:18)
    at HTMLDivElement.eval (manualColumnResize.mjs:535:23)
    at HTMLDivElement.callbackProxy (eventManager.mjs:80:18)
```
The `onMouseOver` handler should only be attached to the Handsontable header, not any other table header.

A related issue is that any `th` in data must have `style="visibility: visible"` because most Handsontable styles leak out – the selectors in CSS are written e.g. like `.handsontable .ht_master tr th`, which affects all `th` elements. All Handsontable core style selectors should instead be specific like `.handsontable .ht_master table.htCore > thead > tr > th` so as not to affect anything else.
### Steps to reproduce

1. Open the demo below
2. Do a bit of scrolling up and down (the error doesn't seem to happen after the initial render)
3. Hover over the table header in the "Progress" column.

### Demo

https://codesandbox.io/s/handsontable-javascript-data-grid-hello-world-app-12-3-3-forked-y2p40w

The demo above is just the demo from "Get started with Handsontable" docs (https://codesandbox.io/s/handsontable-javascript-data-grid-hello-world-app-12-3-3-6rf1pz), I've only added the `tableInDataRenderer` function and used it and added `manualColumnResize: true` to config.

### Your environment
* Handsontable version: 12.3.3
* Browser name and version: Chrome 112
* Operating system: Windows 10

A related issue: #9317

**Comment by AMBudnik** [comment]
Thank you for sharing the issue report @rfricz 

You are right once we use `manualColumnResize` we cannot build a `<table>` inside a cell.

I've simplified the demo and checked if it's not the Codebsanodx fault https://jsfiddle.net/mw96cohg/2/

**Comment by AMBudnik** [comment]
I have great news, @rfricz 

This issue is fixed in Handsontable v14.1.0 released today.

I highly encourage you to read more about the changes in our blog post at https://handsontable.com/blog/handsontable-14-1-0-typescript-ssr-improvements
